pgbedrock/core_configure.py

```diff
@@ -131,24 +136,93 @@ def run_password_sql(cursor, all_password_sql_to_run):
         common.fail(msg=common.FAILED_QUERY_MSG.format(query, e))
 
 
-def verify_spec(spec):
-    """ Ensure keywords make sense, particularly because it's easy to use singulars
-    instead of plurals (e.g. 'table' instead of 'tables') """
+def verify_scheme(spec):
+    """Checks spec for schema errors."""
     assert isinstance(spec, dict)
+    error_messages = []
 
     schema = yaml.load(SPEC_SCHEMA_YAML)
     v = cerberus.Validator(schema)
-    error_messages = []
     for rolename, config in spec.items():
         if not config:
             continue
-
         v.validate(config)
         for field, err_msg in v.errors.items():
             error_messages.append(VALIDATION_ERR_MSG.format(rolename, field, err_msg[0]))
 
-    if error_messages:
-        common.fail('\n'.join(error_messages))
+    return error_messages
+
+
+def check_for_multi_schema_owners(spec):
+    """Checks spec for schema with multiple owners."""
+    assert isinstance(spec, dict)
+    error_messages = []
+
+    globally_owned_schemas = defaultdict(list)
+    for role, config in spec.items():
+        if config and config.get('has_personal_schema'):
+            # indicates a role has a personal schema with its same name
+            globally_owned_schemas[role].append(role)
+        # eventually pgbedrock will support more than schema ownership, this will break
+        if config and config.get('owns') and config['owns'].get('schemas'):
+            role_owned_schemas = config['owns']['schemas']
+            for role_schema in role_owned_schemas:
+                globally_owned_schemas[role_schema].append(role)
+    for role_schema in globally_owned_schemas:
+        if len(globally_owned_schemas[role_schema]) > 1:
+            owners = ", ".join(globally_owned_schemas[role_schema])
+            error_messages.append(MULTIPLE_SCHEMA_OWNER_ERR_MSG.format(role_schema, owners))
+
+    return error_messages
+
+
+def check_read_write_obj_references(spec):
+    """Verifies objects aren't defined in both read and write privileges sections."""
+    assert isinstance(spec, dict)
+    error_messages = []
+
+    multi_refs = defaultdict(dict)
+    for role, config in spec.items():
+        if config and config.get('privileges'):
+            for obj in config['privileges']:
+                try:
+                    reads = set(config['privileges'][obj]['read'])
+                    writes = set(config['privileges'][obj]['write'])
+                    duplicates = reads.intersection(writes)
+                    if duplicates:
+                        multi_refs[role][obj] = list(duplicates)
+                except KeyError:
+                    continue
+    if multi_refs:
+        multi_ref_strings = ["%s: %s" % (k, v) for k, v in multi_refs.items()]
+        multi_ref_err_string = ", ".join(multi_ref_strings)
```

Comment:
It might make more sense to join these with `\n\t` since then the output (given the change proposed above to `OBJECT_REF_READ_WRITE_ERR`) would look something like:
```
Spec error: objects have been unnecessarily given both read and write privileges. pgbedrock automatically grants read access when write access is requested:
    zmarine: {'tables': ['dmart.website_dm', 'dmart.employee_dm']}
    jshiver: {'tables': ['sqsp.websites', 'ods.stripe_charge_ft']}
    rmaxwell: {'schemas': ['ods', 'sqsp']}
```